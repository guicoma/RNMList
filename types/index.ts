export type CharacterStatus = 'Alive' | 'Dead' | 'unknown';
export type CharacterLocation = {
    name: string,
    url: string
}
export type Character = {
    id: number,
    name: string,
    status: CharacterStatus,
    species: string,
    type: string,
    gender: string,
    origin: CharacterLocation,
    location: CharacterLocation,
    image: string,
    episode: string[],
    url: string,
    created: string,
};
export type PagedCharacter = {
    info: {
        count: number,
        pages: number,
        next: string | null,
        prev: string | null
    },
    results : Character[]
}


export type Episode = {
    id:	number,
    name: string,
    air_date: string,
    episode: string,
    characters:	string[],
    url: string,
    created: string,
};

export type Location = {
    id:	number,
    name: string,
    type: string,
    dimension: string,
    residents: string[],
    url: string,
    created: string,
};

export type LocationParams = {
    id: string,
};