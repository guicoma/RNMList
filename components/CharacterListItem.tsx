import { Character } from "@/types";
import { Image, StyleSheet, View } from "react-native";
import { ThemedText } from "./ThemedText";

export default function CharacterListItem({character}: {character: Character}) {
    return (
        <View style={styles.item}>
            <Image source={{uri: character.image}}
                   style={styles.image}
            />
            <ThemedText>{character.name}</ThemedText>
        </View>
    )
}

const styles = StyleSheet.create({
    item : {
        flexDirection: 'row',
        flex: 1,
        alignItems: 'center',
        marginHorizontal: 8,
    },
    image: {
        width: 16,
        height: 16,
        borderRadius: 25,
        marginHorizontal: 4,
    }
})