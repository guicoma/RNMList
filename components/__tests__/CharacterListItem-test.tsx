import * as React from 'react';
import renderer from 'react-test-renderer';

import CharacterListItem from '../CharacterListItem';
import { Character } from '@/types';

const character: Character = {
    "id": 1,
    "name": "Rick Sanchez",
    "status": "Alive",
    "species": "Human",
    "type": "",
    "gender": "Male",
    "origin": {
        "name": "Earth",
        "url": "https://rickandmortyapi.com/api/location/1"
    },
    "location": {
        "name": "Earth",
        "url": "https://rickandmortyapi.com/api/location/20"
    },
    "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
    "episode": [
        "https://rickandmortyapi.com/api/episode/1",
        "https://rickandmortyapi.com/api/episode/2",
    ],
    "url": "https://rickandmortyapi.com/api/character/1",
    "created": "2017-11-04T18:48:46.250Z"
};

it(`renders correctly`, () => {
  const tree = renderer.create(<CharacterListItem character={character} />).toJSON();

  expect(tree).toMatchSnapshot();
});
