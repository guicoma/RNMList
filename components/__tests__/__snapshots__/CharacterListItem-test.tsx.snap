// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "marginHorizontal": 8,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      }
    }
    style={
      {
        "borderRadius": 25,
        "height": 16,
        "marginHorizontal": 4,
        "width": 16,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Rick Sanchez
  </Text>
</View>
`;
